src/pages/index.js

```diff
@@ -92,118 +171,117 @@ enableValidation(settings);
 
 formValidators["profile-form"].resetValidation();
 formValidators["card-form"].resetValidation();
-// formValidators["profile-image-form"].resetValidation();
 
 // ! EVENT HANDLERS
 
 // Profile edit handler
 function handleProfileEditSubmit(profileData) {
-  user.setUserInfo({
-    title: profileData.title,
-    subtitle: profileData.subtitle,
-  });
-  profileEditPopup.close();
+  profileEditPopup.renderLoading(true);
+  api
+    .updateProfileInfo({
+      name: profileData.title,
+      description: profileData.subtitle,
+    })
+    .then((updatedData) => {
+      user.setUserInfo(updatedData);
+      profileEditPopup.close();
+    })
+    .catch((err) => console.error(`Failed to update profile: ${err}`))
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+    });
 }
 
 // Add card handler
-function handleAddCardSubmit(newCardData) {
-  const name = newCardData.title;
-  const alt = newCardData.title;
-  const link = newCardData.image;
-  section.addItem(createCard({ name, alt, link }));
-  addCardPopup.close();
-  formValidators["card-form"].disableButton();
+function handleAddCardSubmit(inputValues) {
+  addCardPopup.renderLoading(true);
+  api
+    .addCard({ name: inputValues.title, link: inputValues.image })
+    .then((newCardData) => {
+      const newCard = createCard(newCardData);
+      cardSection.addItem(newCard.getCardElement());
```

Comment:
**Could be improved**

You can make a universal function for adding a card into the section using different methods: `append`, `prepend`, or anything else:

```js

// The function accepts a card object and a method of adding to the section
// The method is initially `addItem`, but you can pass `appendItem` 
function renderCard(item, method = "addItem") {
  
  const cardElement = createCard(item);
  // Add the card into the section using the method
  cardSection[ method ](cardElement);
}



```